env_store.go

```diff
@@ -32,13 +32,13 @@ import (
 	"github.com/bank-vaults/secret-init/pkg/provider/vault"
 )
 
-var supportedProviders = []string{
-	file.ProviderName,
-	vault.ProviderName,
-	bao.ProviderName,
-	aws.ProviderName,
-	gcp.ProviderName,
-	azure.ProviderName,
+var ProviderTypes = []provider.Provider{
```

Comment:
these are actual providers, not just types, perhaps better to expose as `providers`